### ISSUE TITLE
Not compatible with Magento 2.3.3

### ISSUE THREAD

**Issue by peterjaap** [kiwicommerce/magento2-enhanced-smtp#6]
When also using `KiwiCommerce\CronScheduler`;

```
Run KiwiCommerce\CronScheduler\Controller\Adminhtml\Cron\Sendemail::execute PHP Fatal error:  Uncaught Error: __clone method called on non-object in /data/clientname/magento2/vendor/magento/framework/Mail/Template/TransportBuilder.php:338
Stack trace:
#0 /data/clientname/magento2/vendor/kiwicommerce/module-cron-scheduler/Controller/Adminhtml/Cron/Sendemail.php(230): Magento\Framework\Mail\Template\TransportBuilder->getTransport()
#1 /data/clientname/magento2/vendor/kiwicommerce/module-cron-scheduler/Controller/Adminhtml/Cron/Sendemail.php(131): KiwiCommerce\CronScheduler\Controller\Adminhtml\Cron\Sendemail->sendEmailStatus(Array, Array, Array)
#2 /data/clientname/magento2/generated/code/KiwiCommerce/CronScheduler/Controller/Adminhtml/Cron/Sendemail/Interceptor.php(24): KiwiCommerce\CronScheduler\Controller\Adminhtml\Cron\Sendemail->execute()
#3 [internal function]: KiwiCommerce\CronScheduler\Controller\Adminhtml\Cron\Sendemail\Interceptor->execute(Object(Magento\Cron\Model\Schedule))
#4 phar:///usr/local/bin/n98-magerun2/src/N98/Magento/Command/System/Cron/RunCommand.php(60): call_user_func(Array, Object( in /data/clientname/magento2/vendor/magento/framework/Mail/Template/TransportBuilder.php on line 338

Fatal error: Uncaught Error: __clone method called on non-object in /data/clientname/magento2/vendor/magento/framework/Mail/Template/TransportBuilder.php:338
Stack trace:
#0 /data/clientname/magento2/vendor/kiwicommerce/module-cron-scheduler/Controller/Adminhtml/Cron/Sendemail.php(230): Magento\Framework\Mail\Template\TransportBuilder->getTransport()
#1 /data/clientname/magento2/vendor/kiwicommerce/module-cron-scheduler/Controller/Adminhtml/Cron/Sendemail.php(131): KiwiCommerce\CronScheduler\Controller\Adminhtml\Cron\Sendemail->sendEmailStatus(Array, Array, Array)
#2 /data/clientname/magento2/generated/code/KiwiCommerce/CronScheduler/Controller/Adminhtml/Cron/Sendemail/Interceptor.php(24): KiwiCommerce\CronScheduler\Controller\Adminhtml\Cron\Sendemail->execute()
#3 [internal function]: KiwiCommerce\CronScheduler\Controller\Adminhtml\Cron\Sendemail\Interceptor->execute(Object(Magento\Cron\Model\Schedule))
#4 phar:///usr/local/bin/n98-magerun2/src/N98/Magento/Command/System/Cron/RunCommand.php(60): call_user_func(Array, Object( in /data/clientname/magento2/vendor/magento/framework/Mail/Template/TransportBuilder.php on line 338
```

**Comment by peterjaap** [comment]
I applied this patch; https://community.magento.com/t5/Magento-DevBlog/Backward-incompatible-Changes-in-the-Mail-Library-for-Magento-2/ba-p/144787

But same error unfortunately.

**Comment by peterjaap** [comment]
Ok found it. The preference for `TransportBuilder` you introduced (`SendEmailLog`) returns itself instead of (the result of) the parent method in `prepareMessage()`.

Since that method doesn't do anything to the transport itself but merely logs it, it can return the original transport;

Line 339 change to `$result = parent::prepareMessage();`.

Line 373 change to `return isset($result) ? $result : parent::prepareMessage();`.

That fixes it.

**Comment by kiwi-commerce** [comment]
Hello @peterjaap ,

Thanks for your contribution.
We have applied this fix and released in the extension version 1.0.7.

Thank you.
